<launch>

	<!-- see cfg/mono.cfg for detail -->  
	<arg name="camera_id"		default="0" />
	<arg name="auto_exposure"	default="false" />
	<arg name="auto_gain"		default="true" />
	<arg name="color"		default="6" /> <!-- 0:MONO8, 1:MONO16, 2:YUV, 3:YCbCr, 4:BGR5, 5:BGR565, 6:BGR8, 7:BGRA8, 9:BGRY8, 9:RGB8, 10:RGBA8, 11, RGBY8 -->
	<arg name="config_path"		default="" />
	<arg name="exposure_time"	default="85" />
	<arg name="force_streaming"	default="false" />
	<arg name="frame_id"		default="/camera" />
	<arg name="frame_rate"		default="40" />
	<arg name="gain"		default="100" />
	<arg name="gain_boost"		default="false" />
	<arg name="hardware_gamma"	default="true" />
	<arg name="pixel_clock"		default="237" />
	<arg name="trigger"		default="0" /> <!-- 0:AUTO, 1:HARDWARE_RISING, 2:HARDWARE_FALLING -->
	<arg name="zoom"		default="4" />

	<node pkg="nodelet" type="nodelet" name="ueye_manager" args="manager" /> <!--output="screen"-->

	<node pkg="nodelet" type="nodelet" name="camera_nodelet" args="load ueye/CameraNodelet ueye_manager" >
		<param name="cameraId"		value="$(arg camera_id)" />
		<param name="auto_exposure"	value="$(arg auto_exposure)" />
		<param name="auto_gain"		value="$(arg auto_gain)" />
		<param name="color"		value="$(arg color)" />
		<param name="config_path"	value="$(arg config_path)" />
		<param name="exposure_time"	value="$(arg exposure_time)" />
		<param name="force_streaming"	value="$(arg force_streaming)" />
		<param name="frame_id"		value="$(arg frame_id)" />
		<param name="frame_rate"	value="$(arg frame_rate)" />
		<param name="gain"		value="$(arg gain)" />
		<param name="gain_boost"	value="$(arg gain_boost)" />
		<param name="hardware_gamma"	value="$(arg hardware_gamma)" />
		<param name="pixel_clock"	value="$(arg pixel_clock)" />
		<param name="trigger"		value="$(arg trigger)" />
		<param name="zoom"		value="$(arg zoom)" />
	</node>
	<node pkg="nodelet" type="nodelet" name="framerate_nodelet" args="load ueye/FramerateNodelet ueye_manager"/>

	<node name="marker_detection" pkg="kuri_mbzirc_challenge_1_marker_tracking" type="marker_detection.py" /> <!--output="screen"-->

	<node pkg="cftld_ros" type="cftld_ros_node" name="cftld_ros" output="screen" respawn="true">
    <remap from="image_raw" to="/downward_cam/camera/image" />
    <remap from="init_roi" to="ch1/marker_bb" />
    <remap from="reset" to="cftld/reset" />
    <remap from="track" to="cftld/track" />
    <remap from="debug_image" to="cftld/tracking/image_raw" />

    <param name="tld_config_file" value="$(find cftld_ros)/config/cftld.cfg" />
    <rosparam command="load" file="$(find cftld_ros)/config/default.yaml" />
  </node>

	<node name="image_view1" pkg="image_view" type="image_view" > <!--respawn="false" output="screen"-->
	<remap from="image" to="detection/image_raw" />
	</node>

	<node name="image_view2" pkg="image_view" type="image_view" > <!--respawn="false" output="screen"-->
	<remap from="image" to="cftld/tracking/image_raw" />
	</node>

</launch>
